pkg/operation/botanist/component/etcd/etcd_test.go

```diff
@@ -126,8 +127,8 @@ var _ = Describe("Etcd", func() {
 			Spec: networkingv1.NetworkPolicySpec{
 				PodSelector: metav1.LabelSelector{
 					MatchLabels: map[string]string{
-						"garden.sapcloud.io/role": "controlplane",
-						"app":                     "etcd-statefulset",
+						"gardener.cloud/role": "controlplane",
```

Comment:
Did you verified that the label change is safe for existing etcds? We recently hit an issue where we noticed a deadlock in etcd-druid itself - https://github.com/gardener/etcd-druid/blob/043c88fd71c5787aa40d3d452be77e13718126b3/controllers/etcd_controller.go#L405-L483. Long story short, etcd-druid was not able to recover when `etcd-bootstrap` ConfigMap is missing the expected labels for some reason.
I can see that the etcd-bootstrap still has the label `garden.sapcloud.io/role=controlplane`. Wouldn't etcd-druid enter in the same deadlock - it would first fail to find the ConfigMap with expected labels (due this change), then it would try to create the ConfigMap and would fail with reason "already exists"? Or is this logic refactored and improved in `etcd-druid@v0.9.0`?

Reply:
Nice, thanks for the follow-ups! Good to know that the deadlock issue that we noticed is fixed in `etcd-druid@v0.9.0` and about to be fixed for `etcd-druid@v0.8.x`

Reply:
I changed the labels in a compatible way, so that we can get rid of the `garden.sapcloud.io/role`label in a later release. I totally agree that this is not strictly related to this PR. However, there was a [comment](https://github.com/gardener/gardener/pull/5741#discussion_r842719358) to not use deprecated labels and thus @shreyas-s-rao tried to switch to the new one as well. I hope it's fine now.